service/frontend/workflowHandler.go

```diff
@@ -247,7 +257,7 @@ func (wh *WorkflowHandler) RegisterNamespace(ctx context.Context, request *workf
 		return nil, errShuttingDown
 	}
 
-	if err := wh.versionChecker.ClientSupported(ctx, wh.config.EnableClientVersionCheck()); err != nil {
+	if err := wh.RecordClientVersionAndCheckIfSupported(ctx); err != nil {
```

Comment:
why not do this in interceptor?

Reply:
I was just following the pattern that we already had with the version checker, I'll look into moving it into an interceptor